pulsar-client/src/main/java/org/apache/pulsar/client/impl/schema/StructSchema.java

```diff
@@ -99,6 +108,37 @@ public T decode(byte[] bytes, byte[] schemaVersion) {
         }
     }
 
+    @Override
+    public T decode(ByteBuf byteBuf) {
+        int size = getCanReadSize(byteBuf);
+        return reader.read(tmpBuffer.get(), 0, size);
```

Comment:
I think we should avoid copying the bytes from `ByteBuf` to the tmpBuffer. I believe AVRO provides mechanism to read from an input stream. You can convert the ByteBuf into an input stream.